ir/pipeline/mod.rs

```diff
@@ -104,6 +104,11 @@ typedb_error! {
             signature: Signature,
             return_: ReturnStatement
         ),
+        ReservedKeywordAsIdentifier(
+            9,
+            "A reserved keyword \"{identifier}\" was used as function name",
```

Comment:
Maybe we make it more action oriented: `the reserved keyword ... cannot be used as a ... ` in all the error messages

Reply:
Do we want to have a variant per usage (as function name , as type label, etc.) or is just one enough?
The identifier does contain the span so I could return an offset int the string, if we had the string.